Dockerfile

```diff
@@ -85,30 +89,30 @@ RUN yarn --cwd ui build
 FROM builder as argo-build
 
 ARG IMAGE_OS=linux
-ARG IMAGE_ARCH=amd64
 
 # Perform the build
 WORKDIR /go/src/github.com/argoproj/argo
 COPY . .
 # check we can use Git
 RUN git rev-parse HEAD
 
+ADD hack/image_arch.sh .
 # controller image
-RUN make dist/workflow-controller-linux-${IMAGE_ARCH}
-RUN ["sh", "-c", "./dist/workflow-controller-linux-${IMAGE_ARCH} version | grep clean"]
+RUN ./image_arch.sh && make dist/workflow-controller-linux-${IMAGE_ARCH}
+RUN ./image_arch.sh && ./dist/workflow-controller-linux-${IMAGE_ARCH} version | grep clean
 
 # executor image
-RUN make dist/argoexec-linux-${IMAGE_ARCH}
-RUN ["sh", "-c", "./dist/argoexec-linux-${IMAGE_ARCH} version | grep clean"]
+RUN ./image_arch.sh && make dist/argoexec-linux-${IMAGE_ARCH}
+RUN ./image_arch.sh && ./dist/argoexec-linux-${IMAGE_ARCH} version | grep clean
 
 # cli image
 RUN mkdir -p ui/dist
 COPY --from=argo-ui ui/dist/app ui/dist/app
 # stop make from trying to re-build this without yarn installed
 RUN touch ui/dist/node_modules.marker
 RUN touch ui/dist/app/index.html
-RUN make argo-server.crt argo-server.key dist/argo-linux-${IMAGE_ARCH}
-RUN ["sh", "-c", "./dist/argo-linux-${IMAGE_ARCH} version | grep clean"]
+RUN ./image_arch.sh && make argo-server.crt argo-server.key dist/argo-linux-${IMAGE_ARCH}
```

Comment:
I'm not sure this does what you want - environment variable set in a script don't get exported to the parent shell. Instead on line 88 you could try this:

```
RUN source ./image_arch.sh
```

Reply:
Actually `source` does not exist: `#10 0.243 /bin/sh: 1: source: not found`

But you are right it does not export it outside the subshell

Reply:
try `.` instead of `source`?

Reply:
That works without pipefail, is that fine for you ?

Reply:
my bad `-o pipefail` only works in bash - so you should remove it

Reply:
Done, will push the changes